confignode/src/main/java/org/apache/iotdb/confignode/manager/PermissionManager.java

```diff
@@ -44,7 +65,22 @@ public PermissionManager(Manager configManager, AuthorInfo authorInfo) {
    * @return TSStatus
    */
   public TSStatus operatePermission(AuthorReq authorReq) {
-    return getConsensusManager().write(authorReq).getStatus();
+    TSStatus status = getConsensusManager().write(authorReq).getStatus();
+    if (status.getCode() == TSStatusCode.SUCCESS_STATUS.getStatusCode()) {
+      try {
+        // If the permissions change, clear the cache content affected by the operation
+        if (authorReq.getAuthorType() != ConfigRequestType.CreateUser
+            || authorReq.getAuthorType() != ConfigRequestType.CreateRole) {
+          invalidateCache(authorReq.getUserName(), authorReq.getRoleName());
+        }
+      } catch (IOException | TException e) {
+        logger.error(
+            "Failed to initialize cache,the initialization operation is {}, the error is {}",
+            authorReq.getAuthorType(),
+            e);
+      }
+    }
```

Comment:
The cache invalidation should be operated before write operation, otherwise there may be inconsistency between configNode and some dataNodes which fail to invalidate cache.